Apps/phsignalfx/signalfx_connector.py

```diff
@@ -0,0 +1,469 @@
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        endpoint = "{0}{1}".format('/v2/incident/', param['incidentid'])
+        ret_val, response = self._make_rest_call(
+            endpoint, action_result, headers=self._headers
+        )
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        action_result.add_data(response)
+
+        return action_result.set_status(phantom.APP_SUCCESS, "Incident details fetched successfully")
+
+    def _paginator(self, action_result, user_limit, params, offset=0):
+        incident_list = list()
+
+        while True:
+            params['limit'] = PAGE_SIZE
+            params['offset'] = offset
+
+            ret_val, response = self._make_rest_call(
+                '/v2/incident', action_result, headers=self._headers, params=params
+            )
+            if phantom.is_fail(ret_val):
+                return RetVal(action_result.get_status(), None)
+
+            if not response:
+                return RetVal(phantom.APP_SUCCESS, incident_list)
+
+            try:
+                incident_list.extend(response)
+            except:
+                return RetVal(action_result.set_status(phantom.APP_ERROR, "Failed to parse the response"), None)
+
+            if len(incident_list) >= user_limit:
+                return RetVal(phantom.APP_SUCCESS, incident_list[:user_limit])
+
+            offset += PAGE_SIZE
+            '''
+            This is because the API currently returns a '500' error when queried for an alert with index value greater than 10000
+            '''
+            if offset + PAGE_SIZE > 10000:
+                break
+
+        return RetVal(phantom.APP_SUCCESS, incident_list)
+
+    def _handle_list_incidents(self, param):
+
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+        params = {}
+        params["includeResolved"] = param.get('include_resolved', False)
+        limit = param.get('limit', PAGE_SIZE)
+
+        # Validate 'limit' action parameter
+        ret_val, limit = self._validate_integer(action_result, limit, LIMIT_PARAM_KEY)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        ret_val, incident_list = self._paginator(action_result, limit, params)
+        if phantom.is_fail(ret_val):
+            return action_result.get_status()
+
+        count = 0
+        try:
+            if incident_list:
+                count = len(incident_list)
+                action_result.update_data(incident_list)
+        except Exception as e:
+            err = self._get_error_message_from_exception(e)
+            error_msg = "Failed to parse the response data. {}".format(err)
+            return action_result.set_status(phantom.APP_ERROR, error_msg)
+
```

Comment:
Please add the **total_incidents** summary for the list type of action.

Reply:
Fixed